pwiz_tools/Skyline/Controls/Graphs/SummaryRelativeAbundanceGraphPane.cs

```diff
@@ -219,16 +219,21 @@ public override void UpdateGraph(bool selectionChanged)
 
             var document = GraphSummary.DocumentUIContainer.DocumentUI;
             var graphSettings = GraphSettings.FromSettings();
+            var aggregateOp = GraphValues.AggregateOp.FromCurrentSettings();
 
             // Only create graph data (and recalculate abundances)
             // if settings have changed, the document has changed, or if it
             // is not yet created
+            var dataChanged = false;
             if (_graphData?.GraphPointList == null ||
                 !ReferenceEquals(document, _graphData.Document) ||
-                !Equals(graphSettings, _graphData.GraphSettings))
+                !Equals(graphSettings, _graphData.GraphSettings) ||
+                !Equals(_graphData.AggregateOp, aggregateOp))
             {
                 _graphData = CreateGraphData(document, graphSettings);
+                dataChanged = true;
```

Comment:
"dataChanged" is used to decide whether to reset the graph by calling "UpdateAxes".
I wonder if we could do something smarter than just looking at whether the Document is different.
Maybe we could compare the new graph data to the old graph data and only update the axes if there are more points or the maximum Y value of all of the points has changed.

As it stands, the graph will reset itself when the document changes for reasons that do not affect the layout, such as editing the Note on something in the Targets tree or when the document changes because of a background loader.